.github/workflows/check-pull-request.yml

```diff
@@ -0,0 +1,94 @@
+name: CI_PR_Workflow
+
+on:
+  pull_request:
+    branches:
+      - '**'
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: 'gradle'
+
+      - name: Build with Gradle
+        run: ./gradlew build --no-daemon
+
+  test:
```

Comment:
sonarqube is handling the testing.

Reply:
Yeah, but we also need to ensure that each PR had all the tests passed and to wait SonarQube to verify it.